packages/eslint-plugin/rules/i18n-text-domain.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	getTranslateFunctionName,
+} = require( '../utils' );
+
+/**
+ * Returns the text domain passed to the given translation function.
+ *
+ * @param {string} functionName Translation function name.
+ * @param {Array} args Function arguments.
+ * @return {undefined|*} Text domain argument.
+ */
+function getTextDomain( functionName, args ) {
+	switch ( functionName ) {
+		case '__':
+			return args[ 1 ];
+		case '_x':
+			return args[ 2 ];
+		case '_n':
+			return args[ 3 ];
+		case '_nx':
+			return args[ 4 ];
+		default:
+			return undefined;
+	}
+}
+
+module.exports = {
+	meta: {
+		type: 'problem',
+		schema: [
+			{
+				type: 'object',
+				properties: {
+					// Supports a single string as the majority use case,
+					// but also an array of text domains.
+					allowedTextDomain: {
+						anyOf: [
+							{
+								type: 'array',
+								items: {
+									type: 'string',
+								},
+								uniqueItems: true,
+							},
+							{
+								type: 'string',
+								default: 'default',
+							},
+						],
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+		messages: {
+			invalidValue: "Invalid text domain '{{ textDomain }}'",
+			invalidType: 'Text domain is not a string literal',
+			unnecessaryDefault: 'Unnecessary default text domain',
+			missing: 'Missing text domain',
+			useAllowedValue:
+				'Use one of the whitelisted text domains: {{ textDomains }}',
+		},
+		fixable: 'code',
+	},
+	create( context ) {
+		const options = context.options[ 0 ] || {};
+		const { allowedTextDomain = 'default' } = options;
+		const allowedTextDomains = Array.isArray( allowedTextDomain )
+			? allowedTextDomain
+			: new Array( allowedTextDomain );
```

Comment:
Hm, I thought this might not have been valid (I am accustomed to the form which received `arrayLength`, did not know it could also receive elements). In any case, it seems needlessly more verbose than the array syntax, unless the hope is to be more explicit with demonstrating that it's being cast to an array.

```suggestion
			: [ allowedTextDomain ];
```